tests/test_black.py

```diff
@@ -1289,6 +1318,195 @@ def test_shhh_click(self) -> None:
             except RuntimeError as re:
                 self.fail(f"`patch_click()` failed, exception still raised: {re}")
 
+    @async_test
+    async def test_blackd_request_needs_formatting(self) -> None:
+        if not has_blackd_deps:
+            self.skipTest("blackd's dependencies are not installed")
+            return
+        app = blackd.make_app()
+        async with TestClient(TestServer(app)) as client:  # type: ignore
+            response = await client.post("/", data=b"print('hello world')")
+            self.assertEqual(response.status, 200)
+            self.assertEqual(response.charset, "utf8")
+            self.assertEqual(await response.read(), b'print("hello world")\n')
+
+    @async_test
+    async def test_blackd_request_no_change(self) -> None:
+        if not has_blackd_deps:
+            self.skipTest("blackd's dependencies are not installed")
+            return
+        app = blackd.make_app()
+        async with TestClient(TestServer(app)) as client:  # type: ignore
+            response = await client.post("/", data=b'print("hello world")\n')
+            self.assertEqual(response.status, 204)
+            self.assertEqual(await response.read(), b"")
+
+    @async_test
+    async def test_blackd_request_syntax_error(self) -> None:
+        if not has_blackd_deps:
+            self.skipTest("blackd's dependencies are not installed")
+            return
+        app = blackd.make_app()
+        async with TestClient(TestServer(app)) as client:  # type: ignore
```

Comment:
Looks like mypy is unhappy with the type ignores. Did you put them because flake8-mypy wanted them?

Reply:
When `aiohttp` is installed, mypy says
```
tests/test_black.py:1325:20: error: Call to untyped function "TestClient" in typed context
```
but flake8-mypy still keeps complaining about the unused type: ignore.

When `aiohttp` is not installed, the ignores are unnecessary. Any tips for working this around?

Reply:
Make sure we install the blackd extra on CI, we need it anyway for coverage... and, you know, continuous integration of blackd tests ;-)

Reply:
Yeah I forgot that when splitting it out to an extra.
For the type errors I'll just add aiohttp ignores in mypy.ini 😢

Reply:
How? mypy.ini module-specific sections are about errors raised *within* said modules, not about their usage outside.

Reply:
Well, this did the trick: https://github.com/ambv/black/pull/460/commits/03b77bd0ea3f8803aaf80f8f439f7468cec0c02e